OrbitGl/App.cpp

```diff
@@ -176,15 +176,75 @@ void GLoadPdbAsync( const std::vector<std::wstring> & a_Modules )
     GModuleManager.LoadPdbAsync( a_Modules, [](){ GOrbitApp->OnPdbLoaded(); } );
 }
 
+//TODO: find a better name
+//-----------------------------------------------------------------------------
+void OrbitApp::StartRemoteCaptureBufferingThread()
+{
+    PRINT_FUNC;
+    m_TimerBuffer.clear();
+    m_SamplingCallstackBuffer.clear();
+    m_HashedSamplingCallstackBuffer.clear();
+    m_ContextSwitchBuffer.clear();
+    m_MessageBufferThread = std::make_shared<std::thread>(&OrbitApp::ProcessBufferedCaptureData, this);
+}
+
+//-----------------------------------------------------------------------------
+void OrbitApp::StopRemoteCaptureBufferingThread()
+{
+    PRINT_FUNC;
+    if ( m_MessageBufferThread )
+    {
+        m_MessageBufferThread->join();
+        m_MessageBufferThread = nullptr;
+    }
+}
+
+//-----------------------------------------------------------------------------
+void OrbitApp::ProcessBufferedCaptureData()
+{
+    while (Capture::IsCapturing())
+    {  
+        usleep(20000);
+        ScopeLock lock( m_MessageBufferMutex );
```

Comment:
A single lock protects timers, sampling cs, hashed cs, and context switches.  This causes avoidable serialization/contention.  Can we use one lock per event type?

Reply:
Sure, I'll change it to one lock per event type